scalafmt-core/shared/src/main/scala/org/scalafmt/internal/Router.scala

```diff
@@ -173,6 +173,28 @@ class Router(formatOps: FormatOps) {
           if rightOwner.is[SomeInterpolate] =>
         Seq(Split(Space(style.spaces.inInterpolatedStringCurlyBraces), 0))
 
+      // optional braces: template follows
+      case ft @ FormatToken(
+            _: T.Colon | _: T.KwWith,
+            _,
+            OptionalBracesTmpl(owner)
+          ) if dialect.allowSignificantIndentation =>
+        val indent = style.indent.getSignificant
+        Seq(Split(Newline, 0).withIndent(indent, lastToken(owner), After))
+
+      // optional braces: block follows
+      case ft @ FormatToken(
+            _: T.Equals | _: T.RightArrow | _: T.ContextArrow | _: T.LeftArrow |
```

Comment:
tried avoiding these, since the parser realistically already knows which tokens can be followed by a block without braces. but java kept blowing up, probably because matching works better on case classes and not on `unapply` extractors. so using an extractor only on the `meta` field (which is really sufficient to match on the entire token because it contains the token index).